bin/tflocal

```diff
@@ -25,14 +25,15 @@ import hcl2  # noqa: E402
 
 DEFAULT_REGION = "us-east-1"
 DEFAULT_ACCESS_KEY = "test"
+AWS_ENDPOINT_URL = os.environ.get("AWS_ENDPOINT_URL") or "http://localhost:4655"
 CUSTOMIZE_ACCESS_KEY = str(os.environ.get("CUSTOMIZE_ACCESS_KEY")).strip().lower() in ["1", "true"]
 LOCALHOST_HOSTNAME = "localhost.localstack.cloud"
 S3_HOSTNAME = os.environ.get("S3_HOSTNAME") or f"s3.{LOCALHOST_HOSTNAME}"
 USE_EXEC = str(os.environ.get("USE_EXEC")).strip().lower() in ["1", "true"]
 TF_CMD = os.environ.get("TF_CMD") or "terraform"
 LS_PROVIDERS_FILE = os.environ.get("LS_PROVIDERS_FILE") or "localstack_providers_override.tf"
-LOCALSTACK_HOSTNAME = os.environ.get("LOCALSTACK_HOSTNAME") or "localhost"
-EDGE_PORT = int(os.environ.get("EDGE_PORT") or 4566)
+LOCALSTACK_HOSTNAME = os.environ.get("LOCALSTACK_HOSTNAME") or urlparse(AWS_ENDPOINT_URL).hostname
```

Comment:
I would suggest to reverse the logic here - using the hostname/port from `AWS_ENDPOINT_URL` if configured, and falling back to the deprecated `LOCALSTACK_HOSTNAME`/`EDGE_PORT` configs otherwise.. Thoughts?

Reply:
So should we end up with 3 different cases as with that logic it would make sense to remove the default value from `AWS_ENDPOINT_URL`? Something along these lines:  
If `AWS_ENDPOINT_URL` is configured use that, if not use the `LOCALSTACK_HOSTNAME`, if this fails go with the default value `localhost`. 
Same for the `EDGE_PORT`.